MdeModulePkg/Library/DeviceManagerUiLib/DeviceManager.c

```diff
@@ -619,7 +619,8 @@ CreateDeviceManagerForm(
         TokenHelp = HiiSetString (HiiHandle, 0, String, NULL);
         FreePool (String);
 
-        FormSetGuid = ((EFI_IFR_FORM_SET *)Ptr)->Guid;
+        FormSetGuid = AllocatePool (sizeof (EFI_GUID));
+        CopyMem (FormSetGuid, &((EFI_IFR_FORM_SET *) Ptr)->Guid, sizeof (EFI_GUID));

```

Comment:
Actually there is `CopyGuid (DestinationGuid, SourceGuid)`, which would be nicer to use.